subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c

```diff
@@ -345,8 +343,33 @@ static int prepare_cb_common(struct lll_prepare_param *p)
 	}
 	pdu->cssn = lll->cssn;
 
-	radio_pkt_tx_set(pdu);
+	/* Encryption */
+	if (pdu->len && lll->enc) {
+		uint8_t pkt_flags;
 
+		lll->ccm_tx.counter = payload_count;
+
+		(void)memcpy(lll->ccm_tx.iv, lll->giv, 4U);
+		mem_xor_32(lll->ccm_tx.iv, lll->ccm_tx.iv, access_addr);
+
+		pkt_flags = RADIO_PKT_CONF_FLAGS(RADIO_PKT_CONF_PDU_TYPE_DC,
+						 phy,
+						 RADIO_PKT_CONF_CTE_DISABLED);
+		radio_pkt_configure(RADIO_PKT_CONF_LENGTH_8BIT,
+				    (lll->max_pdu + PDU_MIC_SIZE), pkt_flags);
+		radio_pkt_tx_set(radio_ccm_tx_pkt_set(&lll->ccm_tx, pdu));
+	} else {
+		uint8_t pkt_flags;
+
+		pkt_flags = RADIO_PKT_CONF_FLAGS(RADIO_PKT_CONF_PDU_TYPE_DC,
+						 phy,
+						 RADIO_PKT_CONF_CTE_DISABLED);
+		radio_pkt_configure(RADIO_PKT_CONF_LENGTH_8BIT, lll->max_pdu,
+				    pkt_flags);
+		radio_pkt_tx_set(pdu);
+	}
+
+	/* Setup radio IFS switching */
```

Comment:
Define RADIO_PKT_CONF_PDU_TYPE_BIS and RADIO_PKT_CONF_PDU_TYPE_CIS, and use it to set correct PDU header masking required for CCM specification's additional authentication data.